sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventDataBatch.java

```diff
@@ -100,6 +108,20 @@ public int getSizeInBytes() {
         return this.sizeInBytes;
     }
 
+    /**
+     * Gets the sequence number of the first event in the batch, if the batch was successfully
+     * published by a sequence-aware producer.  If the producer was not configured to apply
+     * sequence numbering or if the batch has not yet been successfully published, this member
+     * will be {@code null}.
+     *
+     * @return the publishing sequence number assigned to the first event in the batch at the time
+     * the batch was successfully published. {@code null} if the producer was not configured to apply
+     * sequence numbering or if the batch has not yet been successfully published.
+     */
+    public Integer getStartingPublishedSequenceNumber() {
```

Comment:
This isn't set until publishing is confirmed as successful, correct?   _(we don't want to represent an interim state to callers when the publishing operation is active)_

Reply:
Yes, EventDataBatch's startingPublishedSequenceNumber is set only if a send is successful. I updated the code to also update EventData's related properties only when a send is successful.